crates/neon-runtime/src/napi/tsfn.rs

```diff
@@ -194,3 +256,119 @@ impl<T> Drop for ThreadsafeFunction<T> {
         };
     }
 }
+
+// The following helpers do not panic and instead use `napi_fatal_error` to crash the
+// process in a controlled way, making them safe for use in FFI callbacks.
+mod no_panic {
```

Comment:
Maybe add to this comment that the `#[track_caller]` use here ensures that any fatal errors correctly report the user code's location rather than the Neon internal code that triggers the fatal error.